modin/engines/base/frame/partition_manager.py

```diff
@@ -1258,3 +1258,16 @@ def binary_operation(cls, axis, left, func, right):
             ]
         )
         return result if axis else result.T
+
+    @classmethod
+    @wait_computations_if_benchmark_mode
```

Comment:
why we want this decorator here? isn't this method already "waiting" for end of computations?

Reply:
`drain_call_queue` just drains all calls stored in the queue but it doesn't wait for the end of computations. That's why we want to use the decorator.

Reply:
now I wonder if `finalize()` should wait for computations, too...

Reply:
To construct a proper Modin DataFrame, which has multiple deferred calls, we need to perform all the calls not be waiting for the end of computations. Otherwise, we keep the stuff synchronous.

Reply:
`finalize` should probably not wait for computations, agree. We could also add a parameter to `finalize`, but I think the decorator approach is a better. Do you agree @vnlitvinov?

Reply:
Decorator is enough I guess.